Library/Homebrew/cask/dsl.rb

```diff
@@ -525,12 +528,26 @@ def livecheckable?
     # NOTE: A warning will be shown when trying to install this cask.
     #
     # @api public
-    def deprecate!(date:, because:, replacement: nil)
+    def deprecate!(date:, because:, replacement: nil, replacement_formula: nil, replacement_cask: nil)
+      if replacement_formula && replacement_cask
+        raise ArgumentError, "replacement_formula and replacement_cask specified!"
+      end
```

Comment:
```suggestion
      if [replacement, replacement_formula, replacement_cask].map(&:presence).compact.length > 1
        raise ArgumentError, "more than one of replacement, replacement_formula and/or replacement_cask specified!"
      end
```
or something? it seems we probably don't want >1 of any.